src/EditorFeatures/CSharpTest/Completion/CompletionProviders/AwaitCompletionProviderTests.cs

```diff
@@ -279,5 +279,231 @@ public async Task TestAwaitInLock_TopLevel()
+        }
+
+        [Fact]
+        public async Task TestDotAwaitSuggestAfterDotDot()
+        {
+            await VerifyKeywordAsync(@"
+using System.Threading.Tasks;
+
+static class Program
+{
+    static async Task Main(Task someTask)
+    {
+        someTask.$$.;
+    }
+}", LanguageVersion.CSharp9);
+        }
+
+        [Fact]
+        public async Task TestDotAwaitSuggestAfterDotBeforeType()
+        {
+            await VerifyKeywordAsync(@"
+using System;
+using System.Threading.Tasks;
+
+static class Program
+{
+    static async Task Main(Task someTask)
+    {
+        someTask.$$
+        Int32 i = 0;
+    }
+}", LanguageVersion.CSharp9);
+        }
+
+        [Fact]
+        public async Task TestDotAwaitSuggestAfterDotBeforeAnotherAwait()
+        {
+            await VerifyKeywordAsync(@"
+using System;
+using System.Threading.Tasks;
+
+static class Program
+{
+    static async Task Main(Task someTask)
+    {
+        someTask.$$
+        await Test();
+    }
+
+    async Task Test() { }
+}", LanguageVersion.CSharp9);
+        }
+
+        [Fact]
+        public async Task TestDotAwaitNotAfterDotOnTaskIfAlreadyAwaited()
+        {
+            await VerifyAbsenceAsync(@"
+using System.Threading.Tasks;
+
+class C
+{
+  async Task F(Task someTask)
+  {
+    await someTask.$$
+  }
+}
+", LanguageVersion.CSharp9);
+        }
+
+        [Fact]
+        public async Task TestDotAwaitNotInLock()
+        {
+            await VerifyAbsenceAsync(@"
+using System.Threading.Tasks;
+
+class C
+{
+  async Task F(Task someTask)
+  {
+    lock(this) { someTask.$$ }
+  }
+}
+", LanguageVersion.CSharp9);
+        }
+
+        [Fact]
+        public async Task TestDotAwaitNotInLock_TopLevel()
+        {
+            await VerifyAbsenceAsync(@"
+using System.Threading.Tasks;
+
+lock(this) { Task.CompletedTask.$$ }
+", LanguageVersion.CSharp9);
+        }
+
+        [Fact]
+        public async Task TestDotAwaitNotInQuery()
+        {
+            await VerifyAbsenceAsync(@"
+using System.Linq;
+using System.Threading.Tasks;
+
+class C
+{
+  async Task F()
+  {
+    var z = from t in new[] { Task.CompletedTask }
+            select t.$$
+  }
+}
+", LanguageVersion.CSharp9);
+        }
+
+        [Fact]
+        public async Task TestDotAwaitNotAfterConditionalAccessOfTaskMembers()
+        {
+            // The conditional access suggests, that someTask can be null.
+            // await on null throws at runtime, so the user should do
+            // if (someTask is not null) await someTask;
+            // or
+            // await (someTask ?? Task.CompletedTask)
+            // Completion should not offer await, because the patterns above would change to much code.
+            // This decision should be revised after https://github.com/dotnet/csharplang/issues/35 
+            // is implemented.
+            await VerifyAbsenceAsync(@"
+using System.Threading.Tasks;
+
+class C
+{
+  async Task F(Task someTask)
+  {
+    someTask?.$$
```

Comment:
can we have some additional cases like:  `a.b?.`  `a?.b`  `a.b?.c.`  thanks!

Reply:
See the next test. `TestDotAwaitNotAfterDotInConditionalAccessChain`. I will turn it into theory and add a property `C Prop => this;`

Reply:
sure.